multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapCache.java

```diff
@@ -94,7 +95,12 @@ public EmbeddedMultimapCache(Cache<K, Collection<V>> cache) {
    public CompletableFuture<Optional<CacheEntry<K, Collection<V>>>> getEntry(K key) {
       requireNonNull(key, "key can't be null");
       return cache.getAdvancedCache().getCacheEntryAsync(key)
-            .thenApply(Optional::ofNullable);
+            .thenApply(entry -> {
+               if (entry == null)
+                  return Optional.empty();
+
+               return Optional.of(entryFactory.create(entry.getKey(), entry.getValue().toSet(), entry.getMetadata()));
```

Comment:
Technically we only have to make a copy if the value was retrieved from a local node, but I won't pick multi map performance at this point :)